web/src/narrow_state.ts

```diff
@@ -222,7 +222,7 @@ export let get_first_unread_info = (
 ): {flavor: "cannot_compute" | "not_found"} | {flavor: "found"; msg_id: number} => {
     const cannot_compute_response: {flavor: "cannot_compute"} = {flavor: "cannot_compute"};
     if (current_filter === undefined) {
-        // we don't yet support the all-messages view
+        // We're in either the inbox or recent conversations view.
         blueslip.error("unexpected call to get_first_unread_info");
         return cannot_compute_response;
     }
```

Comment:
This function is always called from a `MessageListData.filter` when entering a message feed view, which is definitely an actual message list. I think this block should be deleted, and the function declared as taking a `Filter` object passed by the caller always. This is cleanup that @amanagr and I missed after reworking `home_msg_list` to use a `Filter` object.

Reply:
Added a prep commit for this refactor of `get_first_unread_info` to take a declared `Filter` parameter.